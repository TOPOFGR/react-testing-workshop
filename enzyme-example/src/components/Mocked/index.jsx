import React from 'react';
import PropTypes from 'prop-types';

const Mocked = ({ onSubmit }) => {

    console.log('API call');

    const onSubmitHandler = (e) => {
        e.preventDefault();
    
        onSubmit();
    };

    return (
        <div>
            Mocked component

            <form onSubmit={onSubmitHandler}>
                <input type="text" />
            <button>Save</button>
            </form>
        </div>
    );
};

Mocked.propTypes = {
    onSubmit: PropTypes.func.isRequired,
};

export default Mocked;
