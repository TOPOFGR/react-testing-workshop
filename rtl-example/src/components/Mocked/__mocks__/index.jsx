import React from 'react';

const Mocked = ({onSubmit}) => {
    return (
        <>
         <button onClick={onSubmit} type="button">Click</button>   
        </>
    );
};

export default Mocked;