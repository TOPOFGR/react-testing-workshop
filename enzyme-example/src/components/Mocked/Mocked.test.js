import { render, mount } from 'enzyme';

import Mocked from ".";

describe('<Mocked/>', () => { 

    const onSubmit = jest.fn();

    beforeEach(() => {
        jest.resetAllMocks();
    });
    
    it('should render without crashing', () => { 
        const view = render(<Mocked onSubmit={onSubmit}/>);

        expect(view).toMatchSnapshot();

     });
    
    it('should render the correct text', () => { 
        const wrapper = mount(<Mocked onSubmit={onSubmit}/>);

        const title = wrapper.find('div');

        expect(title).toHaveLength(1);

     });
    
    it('should call the onSubmit function', () => { 
        const wrapper = mount(<Mocked onSubmit={onSubmit}/>);

        const input = wrapper.find('input');
        const form = wrapper.find('form');

        input.simulate('change', { target: { value: 'Hello' } })

        form.simulate('submit');

        expect(onSubmit).toHaveBeenCalledTimes(1);

     });
 })