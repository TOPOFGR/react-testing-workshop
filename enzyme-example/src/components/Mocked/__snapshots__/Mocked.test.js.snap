// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Mocked/> should render without crashing 1`] = `
LoadedCheerio {
  "0": <div>
    Mocked component
    <form>
      <input
        type="text"
      />
      <button>
        Save
      </button>
    </form>
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
